src/components/MySpots/MySpots.tsx

```diff
@@ -1,59 +1,62 @@
-import React from 'react';
+import React, { FC, useState, useRef } from 'react';
 import styled from 'styled-components';
-import { HamburgerMenu } from 'global/Layouts/HamburgerMenu/HamburgerMenu';
-import { TopNav } from '../TopNav/TopNav';
-import BoxInSpots from 'assets/BoxInSpots.svg';
-import AddIcon from 'assets/AddIcon.svg';
-import { NavLink } from 'react-router-dom';
+import { Carousel as AntCarousel } from 'antd';
+import { TopNav } from 'components/TopNav/TopNav';
+import { PageWrapperSpaceBetween } from 'components/PageWrapper';
+import { useQuery } from '@apollo/react-hooks';
+import { PersonaCard } from 'components/PersonaCard/PersonaCard';
+import Carousel from 'components/Carousel/Carousel';
+import { gqlEntity } from 'global/graphqls/schema';
+import { Spinner } from 'components/Spinner/Spinner';
+import { Overlay } from 'components/Overlay/Overlay';
 
-const Wrapper = styled.div`
-  display: flex;
-  justify-content: center;
-  padding-top: 32px;
-  width: auto;
-  height: 100vh;
+import { MySpotsWithoutSpots } from './MySpotsWithoutSpots';
+import { Route } from 'react-router-dom';
+import { GET_SPOT, GetSpotType } from '../../global/graphqls/Spot';
+
+const CaruouselItem = styled.div`
+  padding: 0 20px;
 `;
 
-const TextAndImg = styled.div`
+const Wrapper = styled.div`
   display: flex;
   flex-direction: column;
-  align-items: center;
+  height: 70vh;
 `;
 
-const TextUnderImg = styled.h5`
-  opacity: 0.5;
-`;
-const CreateSpot = styled.div`
-  width: 56px;
-  height: 56px;
-  border-radius: 50%;
-  position: fixed;
-  bottom: 66px;
-  right: 18px;
-  background-color: ${props => props.theme.colors.main.primary};
-  display: flex;
-  justify-content: center;
-  align-items: center;
-`;
+export const MySpots: FC = () => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const { loading, data } = useQuery<GetSpotType>(GET_SPOT);
+  const carousel = useRef<AntCarousel>(null);
 
-const CreateIcon = styled.img``;
+  if (loading) {
+    return (
+      <Overlay>
+        <Spinner />
+      </Overlay>
+    );
+  }
+  if (!data) {
```

Comment:
what if looks like data.userSpots = [] ?

Reply:
i changed on that   'if (isEmpty(data?.userSpots) || !data)' like u have